core/utils/files.go

```diff
@@ -170,3 +174,14 @@ func (s FileSize) String() string {
 	str, _ := s.MarshalText()
 	return string(str)
 }
+
+func MakeTestFile(t *testing.T, contents any, fileName string) string {
+	d := t.TempDir()
+	p := filepath.Join(d, fileName)
+
+	b, err := toml.Marshal(contents)
```

Comment:
Oh, this is the problem - you can't write secrets like this via the existing types, by design. It will always just print redacted values. Can we use raw TOML here instead?

Reply:
You can use `package embed` https://pkg.go.dev/embed to make regular files available to the test.